File: frontend/src/types/room.ts
// Типы номеров — соответствуют room-service API.

// Три класса номеров — влияют на привилегии буфета и доп. услуг.
export type RoomClass = 'ORDINARY' | 'MIDDLE' | 'PREMIUM'

// Русские названия для UI.
export const ROOM_CLASS_LABELS: Record<RoomClass, string> = {
  ORDINARY: 'Обычный',
  MIDDLE: 'Средний',
  PREMIUM: 'Премиум',
}

// Tailwind CSS классы для цветовых бейджей класса номера.
export const ROOM_CLASS_COLORS: Record<RoomClass, string> = {
  ORDINARY: 'bg-gray-100 text-gray-700',   // серый = стандарт
  MIDDLE: 'bg-blue-100 text-blue-700',     // синий = средний
  PREMIUM: 'bg-yellow-100 text-yellow-700', // золотой = премиум
}

// Ответ GET /api/rooms/{id} и GET /api/rooms.
export interface Room {
  id: number
  roomNumber: string  // "101", "203" — человекочитаемый номер
  roomClass: RoomClass
  capacity: number    // максимальное кол-во гостей
  pricePerNight: number
  description?: string
}

// Тело запроса POST /api/rooms и PUT /api/rooms/{id} (ADMIN).
export interface RoomRequest {
  roomNumber: string
  roomClass: RoomClass
  capacity: number
  pricePerNight: number
  description?: string
}

File: room-service/src/main/java/com/pethotel/room/entity/Room.java
package com.pethotel.room.entity;

import com.pethotel.common.enums.RoomClass;
import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.List;

// Сущность номера отеля. Связана с RoomUnavailableDate через @OneToMany.
@Entity
@Table(name = "rooms", schema = "room")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Room {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    // unique = true → UNIQUE constraint в PostgreSQL; одинаковые номера комнат запрещены.
    @Column(unique = true, nullable = false)
    private String roomNumber;

    // EnumType.STRING — хранить "ORDINARY"/"MIDDLE"/"PREMIUM" вместо числового индекса.
    @Enumerated(EnumType.STRING)
    @Column(nullable = false)
    private RoomClass roomClass;

    @Column(nullable = false)
    private Integer capacity;

    // BigDecimal — тип для денежных величин: точная арифметика без ошибок округления float/double.
    @Column(nullable = false)
    private BigDecimal pricePerNight;

    private String description;

    // @OneToMany — один номер имеет много "недоступных дат".
    // mappedBy = "room" — владелец связи находится в RoomUnavailableDate (там @JoinColumn).
    //   Hibernate не создаёт дополнительную связующую таблицу — FK лежит в room_unavailable_dates.
    // cascade = ALL — операции с Room каскадируются на даты: удалили Room → удалились и все его даты.
    // orphanRemoval = true — если дату убрали из коллекции Java-кодом, Hibernate её удалит из БД.
    //   Это нужно для blockDates/unblockDates: работаем с коллекцией, не пишем DELETE вручную.
    @OneToMany(mappedBy = "room", cascade = CascadeType.ALL, orphanRemoval = true)
    // @Builder.Default — без этой аннотации @Builder оставил бы поле null вместо пустого списка.
    // Нам нужен ArrayList, чтобы Hibernate мог добавлять/убирать элементы через прокси-коллекцию.
    @Builder.Default
    private List<RoomUnavailableDate> unavailableDates = new ArrayList<>();
}
